### DISCUSSION TITLE
How to get babel config which applies to given file

### DISCUSSION THREAD

**Discussion by amilajack** [babel/babel#11602] [General] [answered]
Is there a way to get all the babel plugins, presets, and other babel options that apply to a given file? My ultimate goal is to determine the preset env targets that apply to a given file. This will help [eslint-plugin-compat](http://github.com/amilajack/eslint-plugin-compat) infer browser targets for a given file if the user isn't using a `.browserslistrc`.

**Comment by nicolo-ribaudo** [comment, marked as answer]
`@babel/core` exposes two functions to load the user config: https://babeljs.io/docs/en/babel-core#loadoptions

You could do something like this:
```js
const presets = babel.loadPartialConfig({ filename: filename }).options.presets.find(isBabelPresetEnvPreset).options.targets
```

However, this will break if
1. The user loads the preset in a not-common way (for example, directly using `require`)
2. The Babel preset is enabled inside another preset

I suggest asking your users to directly use `.browserslistrc`, since the point of having a separate file for that is exactly so that it can be used by different tools.

**Reply by jedwards1211** [reply]
@nicolo-ribaudo I'm trying both `loadConfig` and `loadPartialConfig` and they don't actually config from my `.babelrc.js` at all...

```js
> var babel = require('@babel/core')
undefined
> babel.loadOptionsSync()
{
  cloneInputAst: true,
  babelrc: false,
  configFile: false,
  passPerPreset: false,
  envName: 'development',
  cwd: '/Users/andy/github/clarity',
  root: '/Users/andy/github/clarity',
  plugins: [],
  presets: []
}
```

How do I make it *actually* `loadFullConfig`? Argh!

**Reply by jedwards1211** [reply]
Okay I see that without any options it will do that, it was also not loading anything when I typed `cwd`/`filename` because I mistyped the `cwd` and it didn't exist.  Would be helpful if `loadOptions` etc. error out if the `cwd` doesn't exist instead of just deferring options loading until parse.
